yt_dlp/extractor/funimation.py

```diff
@@ -81,85 +126,96 @@ def _login(self):
 
-        formats = []
-        for source in sources:
-            source_url = source.get('src')
-            if not source_url:
-                continue
-            source_type = source.get('videoType') or determine_ext(source_url)
-            if source_type == 'm3u8':
-                formats.extend(self._extract_m3u8_formats(
-                    source_url, video_id, 'mp4',
-                    m3u8_id='hls', fatal=False))
-            else:
-                formats.append({
-                    'format_id': source_type,
-                    'url': source_url,
-                })
+            current_formats = []
+            for source in sources:
+                source_url = source.get('src')
+                source_type = source.get('videoType') or determine_ext(source_url)
+                if source_type == 'm3u8':
+                    current_formats.extend(self._extract_m3u8_formats(
+                        source_url, display_id, 'mp4', m3u8_id='%s-%s' % (fmt['id'], 'hls'), fatal=False,
+                        note='Downloading %s %s (%s) m3u8 information' % (fmt['uncut'], fmt['lang'], fmt['id'])))
+                else:
+                    current_formats.append({
+                        'format_id': '%s-%s' % (fmt['id'], source_type),
+                        'url': source_url,
+                    })
+                for f in current_formats:
+                    # TODO: Convert language to code
+                    f.update({'language': fmt['lang'], 'format_note': fmt['uncut']})
+                formats.extend(current_formats)
+        self._remove_duplicate_formats(formats)
         self._sort_formats(formats)
 
         return {
-            'id': video_id,
+            'id': compat_str(episode['episodePk']),
             'display_id': display_id,
-            'title': title,
-            'description': description,
-            'thumbnail': self._og_search_thumbnail(webpage),
-            'series': series,
-            'season_number': int_or_none(title_data.get('seasonNum') or _search_kane('season')),
-            'episode_number': int_or_none(title_data.get('episodeNum')),
-            'episode': episode,
-            'subtitles': subtitles,
-            'season_id': title_data.get('seriesId'),
+            'duration': duration,
+            'title': episode['episodeTitle'],
+            'description': episode.get('episodeSummary'),
+            'episode': episode.get('episodeTitle'),
+            'episode_number': int_or_none(episode.get('episodeId')),
+            'episode_id': episode.get('slug'),
+            'season': season.get('seasonTitle'),
+            'season_number': int_or_none(season.get('seasonId')),
+            'season_id': compat_str(season.get('seasonPk')),
+            'series': show.get('showTitle'),
             'formats': formats,
+            'thumbnails': thumbnails,
+            'subtitles': subtitles,
         }
 
     def _get_subtitles(self, url, video_id, display_id):
```

Comment:
Github won't let me leave a comment on line 222 (one below this), so leaving it here.
Line 222 gave me an error due to `video_id `being an int
`ERROR: can only concatenate str (not "int") to str`

Reply:
I havent touched _get_subtitles` yet. It will be removed and subtitle extraction moved to L155

Reply:
@Mevious  just change the player_id line to below.

        player_url = urljoin(url, '/player/' + str(video_id))

That should fix the problem. 👍